core/no_certificate_validation.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+use rustls::internal::msgs::handshake::DigitallySignedStruct;
+use rustls::Certificate;
+use rustls::HandshakeSignatureValid;
+use rustls::RootCertStore;
+use rustls::ServerCertVerified;
+use rustls::ServerCertVerifier;
+use rustls::TLSError;
+use webpki::DNSNameRef;
+
+pub struct NoCertificateVerification {
+  excluded: Vec<String>,
+}
+
+impl NoCertificateVerification {
+  pub fn new(excluded: Vec<String>) -> Self {
+    Self { excluded }
+  }
+}
+
+impl ServerCertVerifier for NoCertificateVerification {
+  fn verify_server_cert(
+    &self,
+    _roots: &RootCertStore,
+    _presented_certs: &[Certificate],
+    dns_name: DNSNameRef<'_>,
+    _ocsp: &[u8],
+  ) -> Result<ServerCertVerified, TLSError> {
+    fn convert_to_string<T: AsRef<str>>(s: T) -> String {
+      s.as_ref().to_owned()
+    }
+    let dns_name: String = convert_to_string(dns_name.to_owned());
```

Comment:
Somehow I could not do dns_name.to_owned().as_ref().to_owned() and cast it to String, that's why the convert_to_string is present - any better ideas?

Reply:
Try this:
```rust
let dns_name: String = (&dns_name.to_owned()).to_owned();
```

Reply:
@littledivy I tried really weird things too, couldn't find a better solution, here is the error log:
```
error[E0308]: mismatched types
  --> core/no_certificate_validation.rs:30:28
   |
30 |     let dns_name: String = (&dns_name.to_owned()).to_owned();
   |                   ------   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `std::string::String`, found struct `DNSName`
   |                   |
   |                   expected due to this
```

Reply:
this works: upgrade rustls to `0.19.1` and -

```suggestion
    let dns_name: &str = dns_name.into();
    let dns_name = dns_name.to_string();
```

Reply:
@littledivy thank you, this works <3